File: java/mix/C05PalindromeType1B.java
public class C05PalindromeType1B {
    // Added comments and white spaces
    public static boolean isPalindrome(String text) {
        StringBuilder cleaned = new StringBuilder();
        
        for (char ch : text.toCharArray()) {
            if (Character.isLetterOrDigit(ch)) {
                    cleaned.append(Character.toLowerCase(ch));
            }
        }

        String cleanedStr = cleaned.toString();
        return cleanedStr.equals(new StringBuilder(cleanedStr).reverse().toString());
    }
}

File: test/javascript/mix/C05PalindromeType1B.js
// Added comments and white spaces
function isPalindrome(text) {
    let cleaned = "";

    for (const ch of text) {
        if (/[a-zA-Z0-9]/.test(ch)) {
            cleaned += ch.toLowerCase();
        }
    }

    const cleanedStr = cleaned;
    return cleanedStr === cleanedStr.split("").reverse().join("");
}

module.exports = { isPalindrome };
